contracts/src/main/kotlin/com/template/contracts/AppointmentRequestContract.kt

```diff
@@ -26,16 +27,37 @@ class AppointmentRequestContract : Contract {
     }
 
     override fun verify(tx: LedgerTransaction) {
-        requireThat {
-            "No inputs should be consumed when issuing a date" using (tx.inputs.isEmpty())
-            "1 reference state should be used" using (tx.references.size == 1)
-            "Only one output state is created" using (tx.outputs.size == 1)
+        val command = tx.findCommand<Commands> { true }
 
-            val out = tx.outputStates[0] as AppointmentRequest
-            "Dates must be of the format dd-MM-yyyy" using (checkDate(out.date))
+        when (command.value) {
```

Comment:
It's a good practice to check not only number if inputs and outputs but also their types and properties.